runtime/observability/telemetry-api/common/src/aws/smithy/kotlin/runtime/telemetry/logging/LogRecordBuilder.kt

```diff
@@ -45,32 +36,22 @@ public interface LogRecordBuilder {
      * @param message the block of code responsible for supplying a log message. `toString()` will be used on the
      * supplied value.
      */
-    public fun setMessage(message: () -> Any)
+    public fun setMessage(message: MessageSupplier)
 
     /**
-     * Set an attribute associated with this log record
+     * Set a key/value pair to associate with this log record
      * @param key the key to use
      * @param value the value to associate with [key]
      */
-    public fun setAttribute(name: String, value: Any)
+    public fun setKeyValuePair(key: String, value: Any)
 
+    // FIXME - revisit this and reconcile with propagators API and/or logging context?
     /**
      * Set the telemetry context to associate with this log record
      * @param context the context to associate
      */
     public fun setContext(context: Context)
 
-    /**
-     * Associate all key/value pairs from [attributes] with this log record
-     * @param attributes the attributes to associate with this log record
-     */
-    public fun mergeAttributes(attributes: Attributes) {
-        attributes.keys.forEach { key ->
-            @Suppress("UNCHECKED_CAST")
-            setAttribute(key.name, attributes[key as AttributeKey<Any>])
-        }
-    }
```

Comment:
Moving away from "attributes" for logging to key/value pairs. This could be an extension function if we need to. I'm still feeling out a few things so it may make it back in at some point.